strings/suffix_tree.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+Suffix tree facilitates searching for patterns in a string
+*/
+
+#include<bits/stdc++.h>
```

Comment:
Please don't use `bits/stdc++.h` as this is quite Linux specific and slows down the compilation process.
Also, in order to use `std::cout` or `std::cin`, you **must** add the `iostream` library.